### PR TITLE
WIP: Added to adas class new adf15 reading capability

### PR DESCRIPTION
**Pull request by cmacmackin**
I thought a pull request would be more appropriate for this than an issue. From @marcosertoli on #24:

> Please check implementation of get_adf15 in ADAS class. Also added ADF15 general datatypes, but meaningful dictionary still needs to be implemented.

I have marked the PR as a Work in Progress (WIP), to indicate that it's not yet ready to be merged.

### CODE REVIEW

File: indica/readers/adas.py [hunk 1]
```diff
@@ -148,6 +150,119 @@ def get_adf11(self, quantity: str, element: str, year: str) -> DataArray:
...
+                if i==0:
+                    nd = int(section_header[2])
+                    nt = int(section_header[3])
+
+                    densities = np.fromfile(f, float, nd, " ")
+                    temperatures = np.fromfile(f, float, nt, " ")
+                    data = np.empty((ntrans, nt, nd))
+                    ttype = [""]*ntrans
+                    isel = np.empty(ntrans)
+                    wlng = np.empty(ntrans)
```
Comment (cmacmackin) [inline indica/readers/adas.py:216]: These should be declared before you enter the loop.

File: indica/readers/adas.py [hunk 2]
```diff
@@ -148,6 +150,119 @@ def get_adf11(self, quantity: str, element: str, year: str) -> DataArray:
...
+
+                m = re.search(r"TYPE =\s*(\S+)", section_header, re.I)
+                pec_type = m.group(1)
+
+                section_header = section_header.split()
+                wavelength = float(section_header[0]) / 10. # (nm)
+
+                if i==0:
+                    nd = int(section_header[2])
+                    nt = int(section_header[3])
```
Comment (cmacmackin) [inline indica/readers/adas.py:209]: Perhaps parse these before entering the loop? that way you could declare data outside of the loop as well and avoid the need for this if/else block.

**Review by cmacmackin — COMMENTED** [review]
I've requested some stylistic changes. The most important ones are making names more descriptive.

As an aside, I remember you said something about deleting some coordinates so that calculations were possible. I'm not sure how you were doing this, but I'd recommend you use the [`DataArray.drop_vars`](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.drop_vars.html#xarray.DataArray.drop_vars) method. This will create a new data array without the specified coordinates while still retaining the old copy which does have the coordinates, in case they're needed again in future.

File: indica/readers/adas.py [hunk 3]
```diff
@@ -148,6 +150,119 @@ def get_adf11(self, quantity: str, element: str, year: str) -> DataArray:
...
+                assert isinstance(m, re.Match)
+                transition.append(m.group(0))
+
+        gen_type = ADF15_GENERAL_DATATYPES[quantity]
+        spec_type = element #ORDERED_ELEMENTS[z]
+        name = f"log_{spec_type}_{gen_type}"
+        attrs = {
+            "datatype": (gen_type, spec_type),
+            "provenance": self.create_provenance(filename, now),
+            "wlng": wlng,
```
[the remark this answers is not in the document]
Comment (marcosertoli) [reply indica/readers/adas.py]: Changed variable names to something more meaningful.
Made wavelength, transition and type coordinates attached to index dimension.

File: indica/readers/adas.py [hunk 4]
```diff
@@ -148,6 +150,119 @@ def get_adf11(self, quantity: str, element: str, year: str) -> DataArray:
...
+            "type": ttype,
+            "transition":transition,
+        }
+
+        print("\n {} ADF15: add info on transition (end of file) \n".format(self.__class__.__name__))
+
+        return DataArray(
+            data * 10**6,
+            coords=[
+                ("isel", isel),
```
[the remark this answers is not in the document]
Comment (marcosertoli) [reply indica/readers/adas.py]: Changed to "index".

File: indica/readers/adas.py (hunk 1 above)
Comment (marcosertoli) [reply indica/readers/adas.py]: Moved them to outside of loop

File: indica/readers/adas.py (hunk 2 above)
Comment (marcosertoli) [reply indica/readers/adas.py:206]: Done

Comment (marcosertoli) [reply indica/readers/adas.py]: Done

**Comment by marcosertoli** [comment]
Added adf15 reading capability, changed log_electron... coordinates to log10_electron as in the adf11 files.

**Pull request merged**